Description: Fix random number generator
 On 64bit systems the random number generator doesn't work as intended.
Author: Drew Parsons
Origin: Debian
Bug: http://bugs.debian.org/457859
Last-Update: 2012-11-16
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: zangband/src/h-type.h
===================================================================
--- zangband.orig/src/h-type.h	2009-02-23 17:36:49.000000000 +1100
+++ zangband/src/h-type.h	2009-02-23 17:44:18.000000000 +1100
@@ -103,6 +103,16 @@
 typedef signed short s16b;
 typedef unsigned short u16b;
 
+/* detect 64 bit GCC */
+#ifdef _LP64
+#ifndef L64
+#define L64 1
+#endif
+#ifndef USE_64B
+#define USE_64B 1
+#endif
+#endif
+
 /* Signed/Unsigned 32 bit value */
 #ifdef L64						/* 64 bit longs */
 typedef signed int s32b;
